src/validate.js

```diff
@@ -1,7 +1,26 @@
-import REGEX from './regex.js';
+import { genericUuid, v1, v3, v4, v5 } from './regex.js';
 
-function validate(uuid) {
-  return typeof uuid === 'string' && REGEX.test(uuid);
+function validateByVersion(uuid, expression) {
+  return typeof uuid === 'string' && expression.test(uuid);
 }
 
-export default validate;
+function isUuid(uuid) {
+  return validateByVersion(uuid, genericUuid);
+}
+export default isUuid;
+
+export function isV1(uuid) {
+  return validateByVersion(uuid, v1);
```

Comment:
Here, and similarly for other `isV#()` methods. 

```suggestion
  return REGEX.test(uuid) && RegExp.$1 === '1';
```

@ctavan: omitting the `typeof uuid === 'string'` condition here since it's not strictly necessary.  `test()` returns `false` for non-strings.  This does mean, however, that `isV#()` would have a slightly different behavior than `validate()`.  Specifically, they could be used to validate an object that casts to a valid uuid string (via `toString()`)